tracing-attributes/tests/fields.rs

```diff
@@ -37,6 +37,36 @@ fn fn_string(s: String) {
 #[instrument(fields(keywords.impl.type.fn = _arg), skip(_arg))]
 fn fn_keyword_ident_in_field(_arg: &str) {}
 
+const CONST_FIELD_NAME: &str = "foo.bar";
+
+#[instrument(fields({CONST_FIELD_NAME} = "quux"))]
+fn fn_const_field_name() {}
+
+const fn get_const_fn_field_name() -> &'static str {
+    "foo.bar"
+}
+
+#[instrument(fields({get_const_fn_field_name()} = "quux"))]
+fn fn_const_fn_field_name() {}
+
+struct FieldNames {}
+impl FieldNames {
+    const FOO_BAR: &'static str = "foo.bar";
+}
+
+#[instrument(fields({FieldNames::FOO_BAR} = "quux"))]
+fn fn_struct_const_field_name() {}
+
+#[instrument(fields({"foo"} = "bar"))]
+fn fn_string_field_name() {}
+
+// TODO: uncomment when the compiler supports const evaluation
```

Comment:
I could actually uncomment this to have a test documenting the current behaviour of constant field name clashes with function arguments if prefered? Just note the compiler would emit a `warning` then about constant `CLASHY_FIELD_NAME` being unused.